File: candy1.py

list1 = [19, 5, 42, 2, 77];
list2 = [23, 15, 59, 4, 17];
def sum_of_smallest_values(arr):
  # 實作程式碼寫在這裡
    return  sorted(arr)[0]+sorted(arr)[1]


print(sum_of_smallest_values(list1)); #印出 7
print(sum_of_smallest_values(list2)); #印出 19

File: JavaScript/candy1.js
// var numbers = [4, 2, 5, 1, 3];
// numbers.sort(function (a, b) {
//   return a - b;
// });
// console.log(numbers);

// 編號：CANDY-001
// 程式語言：JavaScript
// 題目：找出陣列裡最小的兩個值的總和
// 例如：
//   [15, 28, 4, 2, 43] 印出 6
//   [23, 71, 33, 82, 1] 印出 24

function sumOfSmallestValues(arr) {
  // 實作程式碼寫在這裡
  arr.sort((first, last) => {
    return first - last;
  });
  return arr[0] + arr[1];
}

const list1 = [19, 5, 42, 2, 77];
const list2 = [23, 15, 59, 4, 17];

console.log(sumOfSmallestValues(list1)); // 印出 7
console.log(sumOfSmallestValues(list2)); // 印出 19
